Improve Git Workflow page to be more beginner friendly (#202)
ubclaunchpad/docs: resources/git-workflow.md

# 🛶 Git Workflow

Teams at Launch Pad will use the git version control system and GitHub website
to build software collaboratively. This document starts with a "zero-to-hero"
guide to Git and concludes with a recommended GitHub workflow for Launch Pad
teams as well as a basic intro on how to execute that workflow yourself.

If you're familiar with Git, feel free to jump ahead to the [recommended workflow](#recommended-workflow) section.

## Git Zero to Hero

### What is Git?

> Git is fundamentally a *version control system* - a tool that lets you maintain

Reviewer: ```suggestion
Git is a *version control system* - a tool that lets you maintain
```

> a timeline of changes that have occured in your project. Having a version

Reviewer: ```suggestion
a timeline of changes that have occured in your project. Think of it as awesome alternative to saving dozens of copies of a document labelled "report-v1", "report-v2", "report-FINAL", and so on. Having version
```

history is great because you get to freely make changes without ever worrying
about losing prior work.

::: tip

> Git is not the same thing as Git*Hub*. GitHub is a platform that lets you store

Reviewer: Great clarification to include!

(and do other things) your Git-tracked folders (known as repositories) on a
remote server. This lets you easily share these repositories across team
members.
:::

On top of keeping history, Git comes with a set of features that make it truly
great for collaborative work. Git allows you to create *branches*, alternative

> versions of your code, that could later be *merged* back to the main history.

Reviewer: ```suggestion
versions of your code, that could later be *merged* back to the main history.
This allows everyone to make changes to a repository without worrying
about changes that other people are doing!
```

We'll get to this in a bit - let's first talk about how to get your repository
set up.

### Creating a Repository

> TODO: Should this be more conceptual or more tutorial-like?

Reviewer: I would replace this with "What is a repository", since very few people will create a repository

### Saving Changes

Saving changes in a Git repository is a two-step process: first you have to
*stage* your changed files, then you have to *commit* them.

> You can think of a Git commit as a snapshot or milestone along your project
> timeline. Each commit comes with a message that describes what changes that
> milestone introduced.

Reviewer: Hm, a bit unsure of the use of "milestone" - I like "snapshot" better.

```suggestion
You can think of a Git commit as a snapshot along your project
timeline. Each commit comes with a message that describes what changes each
new snapshot introduced, and consists of a "diff" - the difference between
this snapshot and the previous.
```
